mitodl/mit-open
# Save resource prices in a new database model and calculate during ETL/nightly task

### What are the relevant tickets?
Part of https://github.com/mitodl/hq/issues/4759 and https://github.com/mitodl/hq/issues/4830 and 

### Description (What does it do?)
- Adds a `prices` field to `LearningResource` (formerly a property)
- Calculates the value during course/programs ETL (`load_next_start_dates_and_prices`), and a nightly celery task.  New migration calculates the value for courses/programs that can be expected to have non-zero prices.


### How can this be tested?
- If you don't already have courses/programs with prices yet, **on the main branch** ingest some by running the following (after setting required .env variables `XPRO_*` and `MITX_ONLINE_*` based on RC values):
  ```
  ./manage.py backpopulate_mitxonline_data
  ./manage.py backpopulate_xpro_data  
  ```
- Go to http://open.odl.local:8062/search/?certification_type=professional&certification_type=completion, you should get results with prices for certificates.  Take a screenshot.
- Switch to this branch.  Restart containers.  The migration should complete successfully.
- Run `./manage.py update_index --all`
- Refresh the same search url as above.  You should get the same prices back per course as before.  
- Run the same two management commands again.  They should complete successfully, and the prices in the search results should again be the same.
- Run `./manage.py backpopulate_podcast_data` too just to make sure the new code works successfully for resosurces without runs or prices.
